### ISSUE TITLE
Panic on `mdm`

### ISSUE THREAD

**Issue by emilyyyylime** [helix-editor/helix#6626]
### Summary

I couldn't really reproduce this, but I was basically in this situation:
```rs
mod name [{
  // A lot of code...
}]
```

then I deleted the module contents, replacing it with `;` (to achieve `mod name;`). I used rust-analyser to generate `name.rs`, which I then entered with `gd`. In there I pasted the module contents, but of course it was still surrounded by `{}`, so I did `mdm` to remove those braces with the entire file still selected. Helix promptly crashed and I lost my entire module:sob:...

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

Error ought to have happened around here... perhaps around the '1 unsaved buffer(s)' part?
```
2023-04-06T18:26:11.415 helix_lsp::transport [ERROR] err <- "[ERROR hir_ty::mir::lower] MIR lower implementation bug: unsized temporaries    \n"
2023-04-06T18:26:14.982 helix_lsp::transport [ERROR] err <- "[ERROR hir_ty::mir::lower] MIR lower implementation bug: unsized temporaries    \n"
2023-04-06T18:26:27.166 helix_vcs [ERROR] PeelToCommit(
    Unborn {
        name: FullName(
            "refs/heads/master",
        ),
    },
)
2023-04-06T18:26:27.166 helix_vcs [ERROR] failed to open diff base for /path/to/module/name.rs
2023-04-06T18:26:27.166 helix_vcs [ERROR] PeelToCommit(
    Unborn {
        name: FullName(
            "refs/heads/master",
        ),
    },
)
2023-04-06T18:26:27.166 helix_vcs [ERROR] failed to obtain current head name for /path/to/module/name.rs
2023-04-06T18:26:27.214 helix_view::editor [ERROR] editor error: No definition found.
2023-04-06T18:26:27.377 helix_lsp::transport [ERROR] err <- "[ERROR hir_ty::mir::lower] MIR lower implementation bug: unsized temporaries    \n"
2023-04-06T18:27:08.496 helix_vcs [ERROR] PeelToCommit(
    Unborn {
        name: FullName(
            "refs/heads/master",
        ),
    },
)
2023-04-06T18:27:08.496 helix_vcs [ERROR] failed to open diff base for /path/to/module/name.rs
2023-04-06T18:27:08.497 helix_vcs [ERROR] PeelToCommit(
    Unborn {
        name: FullName(
            "refs/heads/master",
        ),
    },
)
2023-04-06T18:27:08.497 helix_vcs [ERROR] failed to obtain current head name for /path/to/module/name.rs
2023-04-06T18:27:08.955 helix_lsp::transport [ERROR] <- ServerError(-32801): content modified
2023-04-06T18:27:08.955 helix_view::editor [ERROR] editor error: Async job failed: protocol error: ServerError(-32801): content modified
2023-04-06T18:27:29.696 helix_vcs [ERROR] PeelToCommit(
    Unborn {
        name: FullName(
            "refs/heads/master",
        ),
    },
)
2023-04-06T18:27:29.696 helix_vcs [ERROR] failed to open diff base for /path/to/module.rs
2023-04-06T18:27:29.696 helix_vcs [ERROR] PeelToCommit(
    Unborn {
        name: FullName(
            "refs/heads/master",
        ),
    },
)
2023-04-06T18:27:29.696 helix_vcs [ERROR] failed to obtain current head name for /path/to/module.rs
2023-04-06T18:27:30.864 helix_lsp::transport [ERROR] <- ServerError(-32801): content modified
2023-04-06T18:27:30.864 helix_view::editor [ERROR] editor error: Async job failed: protocol error: ServerError(-32801): content modified
2023-04-06T18:28:10.155 helix_vcs [ERROR] PeelToCommit(
    Unborn {
        name: FullName(
            "refs/heads/master",
        ),
    },
)
2023-04-06T18:28:10.155 helix_vcs [ERROR] failed to open diff base for /path/to/module/name.rs
2023-04-06T18:28:10.155 helix_vcs [ERROR] PeelToCommit(
    Unborn {
        name: FullName(
            "refs/heads/master",
        ),
    },
)
2023-04-06T18:28:10.155 helix_vcs [ERROR] failed to obtain current head name for /path/to/module/name.rs
2023-04-06T18:28:15.406 helix_view::editor [ERROR] editor error: Surround pair not found around all cursors
2023-04-06T18:28:17.558 helix_view::editor [ERROR] editor error: Surround pair not found around all cursors
2023-04-06T18:28:46.925 helix_view::editor [ERROR] editor error: Surround pair not found around all cursors
2023-04-06T18:28:48.670 helix_view::editor [ERROR] editor error: 1 unsaved buffer(s) remaining: ["module/name.rs"]
2023-04-06T18:28:49.561 helix_view::editor [ERROR] editor error: 1 unsaved buffer(s) remaining: ["module/name.rs"]
2023-04-06T18:31:48.288 helix_vcs [ERROR] PeelToCommit(
    Unborn {
        name: FullName(
            "refs/heads/master",
        ),
    },
)
2023-04-06T18:31:48.288 helix_vcs [ERROR] failed to open diff base for /path/to/module.rs
2023-04-06T18:31:48.289 helix_vcs [ERROR] PeelToCommit(
    Unborn {
        name: FullName(
            "refs/heads/master",
        ),
    },
)
2023-04-06T18:31:48.289 helix_vcs [ERROR] failed to obtain current head name for /path/to/module.rs
2023-04-06T18:31:49.463 helix_lsp::transport [ERROR] <- ServerError(-32801): content modified
2023-04-06T18:31:49.463 helix_view::editor [ERROR] editor error: Async job failed: protocol error: ServerError(-32801): content modified
2023-04-06T18:33:00.972 helix_lsp::transport [ERROR] <- ServerError(-32801): content modified
2023-04-06T18:33:00.976 helix_view::editor [ERROR] editor error: Async job failed: protocol error: ServerError(-32801): content modified
2023-04-06T18:33:18.389 helix_view::editor [ERROR] editor error: no such command: 'W'
2023-04-06T18:36:23.074 helix_vcs [ERROR] PeelToCommit(
    Unborn {
        name: FullName(
            "refs/heads/master",
        ),
    },
)
2023-04-06T18:36:23.074 helix_vcs [ERROR] failed to open diff base for /path/to/module.rs
2023-04-06T18:36:23.075 helix_vcs [ERROR] PeelToCommit(
    Unborn {
        name: FullName(
            "refs/heads/master",
        ),
    },
)
2023-04-06T18:36:23.075 helix_vcs [ERROR] failed to obtain current head name for /path/to/module.rs
2023-04-06T18:36:23.542 helix_lsp::transport [ERROR] <- ServerError(-32801): content modified
2023-04-06T18:36:23.542 helix_view::editor [ERROR] editor error: Async job failed: protocol error: ServerError(-32801): content modified
2023-04-06T18:36:27.281 helix_lsp::transport [ERROR] <- ServerError(-32801): content modified
2023-04-06T18:36:27.282 helix_view::editor [ERROR] editor error: Async job failed: protocol error: ServerError(-32801): content modified
2023-04-06T18:36:39.140 helix_vcs [ERROR] PeelToCommit(
    Unborn {
        name: FullName(
            "refs/heads/master",
        ),
    },
)
2023-04-06T18:36:39.140 helix_vcs [ERROR] failed to open diff base for /path/to/module/name.rs
2023-04-06T18:36:39.140 helix_vcs [ERROR] PeelToCommit(
    Unborn {
        name: FullName(
            "refs/heads/master",
        ),
    },
)
2023-04-06T18:36:39.140 helix_vcs [ERROR] failed to obtain current head name for /path/to/module/name.rs
2023-04-06T18:36:43.977 helix_view::editor [ERROR] editor error: Surround pair not found around all cursors
2023-04-06T18:36:45.576 helix_view::editor [ERROR] editor error: Surround pair not found around all cursors
2023-04-06T18:36:46.714 helix_view::editor [ERROR] editor error: Surround pair not found around all cursors
2023-04-06T18:36:51.570 helix_view::editor [ERROR] editor error: Surround pair not found around all cursors
2023-04-06T18:36:54.141 helix_view::editor [ERROR] editor error: Surround pair not found around all cursors
2023-04-06T18:46:10.173 helix_view::editor [ERROR] editor error: no such command: 'log'
```

</details>

### Platform

Linux

### Terminal Emulator

Kitty

### Helix Version

helix 22.12 (583f6a13)

**Comment by pascalkuthe** [comment]
please provide a reporduction case that can reliably reproduce this issue, otherwise this is not actionable.

**Comment by pascalkuthe** [comment]
Likely closed by #9796, but I can't tell si ce no reproduction case was provided
